PlatformRequirements.md

```diff
@@ -0,0 +1,144 @@
+ 
+ 
+- has an `addPlugin` method
+    - PlatformApi.prototype.addPlugin = function (plugin, installOptions) {};
+    - The `addPlugin` method should install a plugin into a platform. It should handle all the non-www files shipped by plugin (sources, libs, assets, js-files) and accept a PluginInfo instance that represents the plugin that will be installed and an options object. It cannot resolve the dependencies of a plugin.
+    - plugin : This is a PluginInfo instance that should represent the plugin that will be installed. It is expected to accept a plugin spec that should be one of the following:
+        - valid plugin id that can be resolved through either cordova plugin registry or npm: 'org.apache.cordova.globalization', 'cordova-plugin-globalization'
+        - valid npm identifier, that resolves to valid plugin : cordova-plugin-globalization@1.0.0
+        - git url, that points to a repo with a valid plugin: http://github.com/apache/cordova-plugin-globalization.git#r.1.0.0
+        - path to local repo of valid plugin: /my/cordova/repositories/cordova-plugin-globalization
+    - installOptions : This is an options object with the following possible options:
+        - installOptions.link : This is a flag that should specify that plugin sources will be symlinked to app's directory instead of copying (if possible).
+        - installOptions.variables : This is an object that should represent variables that will be used to install a plugin.
+    - The `addPlugin` method must return a promise either fulfilled or rejected with a CordovaError instance.
+- has a `removePlugin` method
+    - PlatformApi.prototype.removePlugin = function (plugin) {};
+    - The `removePlugin` method should remove an installed plugin from a platform. It should accept a PluginInfo instance that represents the plugin that will be removed and an options object.
+    - Note: Since this method accepts the PluginInfo instance as an input parameter, instead of a plugin id, the caller should take care of managing and storing the PluginInfo instances for future uninstalls.
+    - plugin : This is a PluginInfo instance that should represent the plugin that will be uninstalled. It is expected to accept a plugin spec that should be one of the following:
+        - valid plugin id that can be resolved through either cordova plugin registry or npm: 'org.apache.cordova.globalization', 'cordova-plugin-globalization'
+        - valid npm identifier, that resolves to valid plugin : cordova-plugin-globalization@1.0.0
+        - girl url, that points to a repo with a valid plugin: http://github.com/apache/cordova-plugin-globalization.git#r.1.0.0
+        - path to local repo of valid plugin: /my/cordova/repositories/cordova-plugin-globalization
+    - The `removePlugin` method must return a promise either fulfilled or rejected with a CordovaError instance.
+ 
+- CLI work flow integration
```

Comment:
Perhaps it is worth linking to / expanding on the two different kinds of workflows cordova supports? I.e. CLI-based workflow (supporting multiple platforms) or single-platform-based workflow that purely leverages the platform API. e.g. the [top of the Android guide](https://cordova.apache.org/docs/en/latest/guide/platforms/android/index.html) describes there are "platform centered shell tools" or "cross-platform Cordova CLI" workflows.